deploy/cpp_infer/src/main.cpp

```diff
@@ -67,6 +68,13 @@ int main(int argc, char **argv) {
                      config.use_mkldnn, config.use_zero_copy_run,
                      config.char_list_file);
 
+#ifdef USE_MKL
+#pragma omp parallel
+  for (auto i = 0; i < 10; i++) {
+    assert(config.cpu_math_library_num_threads == omp_get_num_threads());
+  }
+#endif
+
```

Comment:
What is the purpose of that code?

Reply:
I think it is better to keep this asset. Because before, MKL was not linked properly, but it does not report any error. Developers and users don't know if it is linked properly. In the code where MKL is used, it is only running on 1 thread, omp_get_num_threads()=1, even config.cpu_math_library_num_threads is set to 8.

Reply:
I have doubts about whether this check should be an assert here. In my opinion a user should be informed with a warning rather than prevented from running Paddle when the number of threads is not appropriate. It would be good to have this assert in a test, not in this file.
@luotao1 , what do you think about it?
Other than that LGTM.

Reply:
I agree with @wojtuss

Reply:
Done. How is now?